core/state/database.go

```diff
@@ -44,7 +46,7 @@ type Database interface {
 	OpenTrie(root common.Hash) (Trie, error)
 
 	// OpenStorageTrie opens the storage trie of an account.
-	OpenStorageTrie(stateRoot common.Hash, address common.Address, root common.Hash) (Trie, error)
+	OpenStorageTrie(stateRoot common.Hash, address common.Address, root common.Hash, trie Trie) (Trie, error)
```

Comment:
@rjl493456442 I added this `trie` parameter, which is a reference to the account trie. Since there is only one verkle tree, this is the trick that I use to make sure that storage writes also go to the main tree.